Convert project to use yarn workspaces

dgeb opened the pull request:
This PR converts this project to use yarn workspaces. This work was motivated by persistent CI failures related to yarn usage with the in-repo addons/engines used to test this project. It's unclear why these failures started, but several other fixes were attempted before resorting to this more extreme step.

rwjblue commented on packages/ember-engines/.template-lintrc.js:
```diff
@@ -0,0 +1,6 @@
+'use strict';
+
+module.exports = {
+  // TODO - "Octanify" templates
```
Probably worth making an issue for this

dgeb replied:
Agreed. See #723

rwjblue commented on packages/ember-engines/node_modules/.bin/_mocha:
```diff
@@ -0,0 +1 @@
+../../../../node_modules/mocha/bin/_mocha
```
I don't grok why this is here, can you remove the checked in `packages/ember-engines/node_modules` folder?

rwjblue commented on packages/ember-engines/node_modules/.bin/ember:
```diff
@@ -0,0 +1 @@
+../../../../node_modules/ember-cli/bin/ember
```
Same as above, seems like this file should not be committed (I would expect `yarn install` to regenerated it as needed).

rwjblue commented on packages/ember-engines/node_modules/.bin/ember-source-channel-url:
```diff
@@ -0,0 +1 @@
+../../../../node_modules/ember-source-channel-url/bin/ember-source-channel-url
```
🔪

rwjblue commented on packages/ember-engines/node_modules/.bin/ember-template-lint:
```diff
@@ -0,0 +1 @@
+../../../../node_modules/ember-template-lint/bin/ember-template-lint.js
```
✂️

rwjblue commented on packages/ember-engines/node_modules/.bin/eslint:
```diff
@@ -0,0 +1 @@
+../../../../node_modules/eslint/bin/eslint.js
```
🍴

rwjblue commented on packages/ember-engines/node_modules/.bin/eslint-config-prettier-check:
```diff
@@ -0,0 +1 @@
+../../../../node_modules/eslint-config-prettier/bin/cli.js
```
🥒

rwjblue commented on packages/ember-engines/node_modules/.bin/lerna-changelog:
```diff
@@ -0,0 +1 @@
+../../../../node_modules/lerna-changelog/bin/cli.js
```
🔫

rwjblue commented on packages/ember-engines/node_modules/.bin/mocha:
```diff
@@ -0,0 +1 @@
+../../../../node_modules/mocha/bin/mocha
```
🧙

rwjblue commented on packages/ember-engines/node_modules/.bin/npm-run-all:
```diff
@@ -0,0 +1 @@
+../../../../node_modules/npm-run-all/bin/npm-run-all/index.js
```
🪓

dgeb commented:
@rwjblue thanks for the review! Sorry about including those node_modules ... I've removed them and rebased. ✂️ 

I created #723 to track the template modernization work that's needed.

dgeb commented:
Looks like there's a failure related to container teardown that's showing in release, beta, and canary. Will have to investigate.
